Userland/Games/ColorLines/ColorLines.cpp

```diff
@@ -0,0 +1,366 @@
+    auto board_header_size = frame_inner_rect().size();
+    board_header_size.set_height(board_vertical_margin);
+    auto const board_header_rect = Gfx::IntRect { frame_inner_rect().top_left(), board_header_size };
+    painter.fill_rect(board_header_rect, Color::Black);
+
+    auto const text_margin = 8;
+
+    // Draw score
+    auto const score_text = String::formatted("{:05}"sv, m_score);
+    auto text_width { m_score_font->width(score_text) };
+    auto const glyph_height = m_score_font->glyph_height();
+    auto const score_text_rect = Gfx::IntRect { frame_inner_rect().top_left().translated(text_margin),
+        Gfx::IntSize { text_width, glyph_height } };
+    painter.draw_text(score_text_rect, score_text, Gfx::TextAlignment::CenterLeft, text_color);
+
+    // Draw high score
+    auto const high_score_text = String::formatted("{:05}"sv, m_high_score);
+    text_width = m_score_font->width(high_score_text);
+    auto const high_score_text_rect = Gfx::IntRect { frame_inner_rect().top_right().translated(-(text_margin + text_width), text_margin),
+        Gfx::IntSize { text_width, glyph_height } };
+    painter.draw_text(high_score_text_rect, high_score_text, Gfx::TextAlignment::CenterLeft, text_color);
+
+    auto const cell_rect
+        = Gfx::IntRect(frame_inner_rect().top_left(), board_cell_dimension).translated(0, board_vertical_margin);
+
+    // Draw all cells and the selected marble if it exists
+    for (int y = 0; y < MarbleBoard::board_size.height(); ++y)
+        for (int x = 0; x < MarbleBoard::board_size.width(); ++x) {
+            auto const& destination_rect = cell_rect.translated(x * board_cell_dimension.width(), y * board_cell_dimension.height());
+            auto const point = Point { x, y };
+            auto const animation_frame = (m_game_state == GameState::MarbleSelected && m_board->has_selected_marble() && m_board->selected_marble().position() == point ? m_marble_animation_frame : default_marble_animation_frame);
+            paint_cell(painter, destination_rect, m_board->color_at(point), animation_frame);
+        }
+
+    // Draw preview marbles in the board
+    for (auto const& marble : m_board->preview_marbles()) {
+        auto const& point = marble.position();
+        if (m_marble_path.contains(point))
+            continue;
+        auto const& destination_rect = cell_rect.translated(point.x() * board_cell_dimension.width(), point.y() * board_cell_dimension.height());
+        auto animation_frame = animation_frame_marble_generating_start;
+        switch (m_game_state) {
+        case GameState::GameOver:
+            animation_frame = default_marble_animation_frame;
+            break;
+        case GameState::GeneratingMarbles:
+        case GameState::CheckingMarbles:
+            animation_frame = m_marble_animation_frame;
+            break;
+        default:
+            animation_frame = animation_frame_marble_generating_start;
+            break;
```

Comment:
No need for this `default` case if the variable is initialized with this value above :^)